plugins/cache/bap_cache_utils.ml

```diff
@@ -0,0 +1,42 @@
+open Core_kernel
+
+module Filename = Caml.Filename
+
+let open_temp temp_dir =
```

Comment:
this function is unsafe and has a race condition via TOCTOU, use `Filename.open_temp_file`